backend/server.js

```diff
@@ -16,9 +18,136 @@ const app = express();
 app.use(cors());
 app.use(express.json());
 
+const UserSchema = new mongoose.Schema({
+  username: {type: String, unique: true, required: true},
+  password: {type: String, required: true},
+  accessToken: {type: String, default: ()=> crypto.randomBytes(128).toString("hex")}
+})
+
+const User = mongoose.model('user', UserSchema)
+
+// Sign up (create new user) POST
+app.post('/register', async (req, res) => {
+  const {username, password} = req.body;
+  try{
+    const salt = bcrypt.genSaltSync();
+    if (password.length < 8) {
+      res.status(400).json({
+        success: false,
+        response: "Password must be at least 8 characters long"
+      });
+    } else {
+      const newUser = await new User({username: username, password: bcrypt.hashSync(password, salt)}).save();
+      res.status(201).json({
+        success: true,
+        response: {
+          username: newUser.username,
+          accessToken: newUser.accessToken,
+          id: newUser._id
+        }
+      });
+    }
+  }catch (error){
+    res.status(400).json({
+      success: false,
+      response: error
+    });
+  }
+});
+
+//Log in (find user and validate the password) POST
+app.post('/login', async (req, res) =>{
+  const { username, password } = req.body;
+
+  try {
+    const user = await User.findOne({username});
+    if (user && bcrypt.compareSync(password, user.password)) {
+      res.status(200).json({
+        success: true,
+        response: {
+          username: user.username,
+          id: user._id,
+          accessToken: user.accessToken
+        }
+      });
+    } else {
+      res.status(400).json({
+        success: false,
+        response: "Credentials didn't match"
+      });
+    }
+  } catch (error) {
+    res.status(500).json({
+      success: false,
+      response: error
+    });
+  }
+});
+
+//Middleware that checks accessTokens that are created when a user is registered
+const authenticateUser = async (req, res, next) => {
+  const accessToken = req.header("Authorization");
+  try {
+    const user = await User.findOne({accessToken: accessToken});
+    if (user) {
+      next();
+    } else {
+      res.status(401).json({
+        response: "Please log in",
+        success: false
+      })
+    }
+  } catch (error) {
+    res.status(400).json({
+      response: error,
+      success: false
+    })
+  }
+}
+
+// const MovieSchema = new mongoose.Schema({
+//   Year: {type: Number},
+//   Title: {type: String},
+//   Rated: {type: String},
+//   Director: {type: String},
+//   Released: {type: String},
+//   Writer: {type: String},
+//   Production: {type: String},
+//   Actors: {type: String},
+//   Runtime: {type: String},
+//   Awards: {type: String},
+//   ImdbVotes: {type: String},
+//   ImdbRating: {type: Number},
+//   RottenTomatoScore: {type: String},
+//   Metascore: {type: Number},
+// }); 
+
+// const Movie = mongoose.model("Movie", MovieSchema);
```

Comment:
Hopefully you get to continue this project and find a way to apply your initial idea. It would be nice if you could create an extra comment to describe why you kept this extra comment below :)